libs/ml/include/ml/core/node.hpp

```diff
@@ -148,6 +149,9 @@ class Node
     return static_cast<bool>(cached_output_status_ == CachedOutputState::VALID_CACHE);
   }
 
+  fetch::vm::ChargeAmount ForwardCost(std::vector<fetch::math::SizeVector> const &input_shapes);
```

Comment:
naming convention might be better to give function names as ChargeX where X is the exact name of the function being charged for (i.e. ChargeEvaluate) etc.
I propose this because I expect that ultimately the right way to do this work is to figure out the charge of every function (even those like ComputeOutputShape).

Reply:
so for the forward pass it would be ChargeForward right?